### PR TITLE
Poy-v0.5

### PR DESCRIPTION
**Pull request by HrikB**
The frontend implementation for Proof of Yield v0.5. It is an attempt to ship the Proof of Yield in project in pieces. For this first iteration, we want to ship the OETH equivalent of https://ousd.com/proof-of-yield while turning of linking to the subsequent pages.

### CODE REVIEW

File: components/proof-of-yield/ChartDetailsButton.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+interface ChartDetailsButtonProps {
+  onClick: () => void;
+}
+
+const ChartDetailsButton = ({
+  onClick,
+  children,
+}: PropsWithChildren<ChartDetailsButtonProps>) => {
+  const width = useViewWidth();
```
Comment (smitch88) [inline components/proof-of-yield/ChartDetailsButton.tsx:16]: This is probably fine but we should try to just be consistent with using this and tailwind breakpoints. Are they set up the same?

File: components/proof-of-yield/DripperGraph.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  THIRTY_DAY = 30,
+  ONE_YEAR = 365,
+}
+
+const dataX = [];
+const dataY = [];
+
+for (let i = 0; i < 30; i++) {
+  dataX.push(Date.now() + i * 24 * 60 * 60 * 1000);
+  dataY.push(((Math.random() * 100000) % 1500) + 2500);
```
Comment (smitch88) [inline components/proof-of-yield/DripperGraph.tsx:47]: Is this just stubbed out data for now?

File: components/proof-of-yield/TableData.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+const TableData = ({
+  align,
+  className,
+  width,
+  children,
+}: PropsWithChildren<TableDataProps>) => {
+  return (
+    <td
+      align={`${align || "right"}`}
```
Comment (smitch88) [inline components/proof-of-yield/TableData.tsx:18]: For consistency, might make sense to just stick with classnames here, text-right/text-left/text-center?

The <td> align attribute is not supported by HTML 5, its kind of obsolete/deprecated

File: components/proof-of-yield/TableHead.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+const TableHead = ({
+  info,
+  align,
+  className,
+  children,
+}: PropsWithChildren<TableHeadProps>) => {
+  return (
+    <th
+      align={`${align || "right"}`}
```
Comment (smitch88) [inline components/proof-of-yield/TableHead.tsx:20]: Similar thoughts on using `align` for td

File: hooks/index.ts [hunk 5]
```diff
@@ -2,3 +2,7 @@ export { default as useOgv } from "./useOgv";
 export { default as useViewWidth } from "./useViewWidth";
 export { default as useContracts } from "./useContracts";
 export { default as useInterval } from "./useInterval";
+export { default as useChartGradient } from "./useChartGradient";
+export { default as useGetPreviousRoute } from "./useGetPreviousRoute";
+export { default as usePreviousRoute } from "./usePreviousRoute";
```
Comment (smitch88) [inline hooks/index.ts:7]: I wonder if we should start picking up a lib with a full set of these type of hooks like `react-use`?

File: sections/proof-of-yield/DailyYield.tsx [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        {/* Table Body */}
+
+        <tbody className="relative px-6">
+          {stats.map((item, i) => (
+            <tr
+              className="group border-t md:border-t-2 hover:bg-hover-bg border-origin-bg-black /cursor-pointer"
+              key={item.date}
+              // onClick={() => routeToYieldOnDay(item.date)}
+            >
+              <TableData align="left" className="pl-8">
```
Comment (smitch88) [inline sections/proof-of-yield/DailyYield.tsx:110]: As Im looking at this usage, I dont think we should really even create a wrapper for `td` or `th`, might as well just add the elements directly

File: sections/proof-of-yield/DayDripperBanner.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+  return (
+    <Section
+      className={sectionOverrideCss}
+      innerDivClassName="border-t-2 border-origin-bg-grey"
+    >
+      <div className="w-full mt-10 md:mt-20 bg-origin-bg-grey py-10 px-12 flex flex-col md:flex-row rounded-lg items-center">
+        <div>
+          <Typography.Body className="text-xl mb-4">
+            Introducing the OUSD Dripper
```
Comment (smitch88) [inline sections/proof-of-yield/DayDripperBanner.tsx:21]: Marking content changes

File: sections/proof-of-yield/DayStrategyPerformance.tsx [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+  img: string;
+  allocationAbs: number;
+  allocationPct: number;
+  weekApyData: number[][];
+  apy: number;
+  yieldAbs: number;
+  yieldPct: number;
+}
+
+let partialMockStratData: { name: string; protocol: string; img: string }[] = [
```
Comment (smitch88) [inline sections/proof-of-yield/DayStrategyPerformance.tsx:29]: Just marking OUSD related data

File: components/proof-of-yield/ChartDetailsButton.tsx (hunk 1 above)
Comment (HrikB) [reply components/proof-of-yield/ChartDetailsButton.tsx:16]: Yup, they are

File: components/proof-of-yield/DripperGraph.tsx (hunk 2 above)
Comment (HrikB) [reply components/proof-of-yield/DripperGraph.tsx:47]: Yeah this is a component for the dripper page (See https://www.ousd.com/proof-of-yield/dripper) that is not being implemented in this PR.

File: components/proof-of-yield/YieldBoostMultiplier.tsx [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+      </div>
+      <div className="border-x md:border-x-2 mx-4 md:mx-8 border-t-0 border-origin-bg-black px-4 md:px-6 py-4 flex flex-col 2xl:flex-row 2xl:justify-between 2xl:items-center">
+        <div>
+          <Typography.Body className="inline mr-3">x</Typography.Body>
+          <Typography.Body className="inline mr-1">2.63</Typography.Body>
+          <Typography.Body2 className="inline">Boost</Typography.Body2>
+        </div>
+        <div className="max-w-[60%]">
+          <Typography.Body3 className="text-table-title text-sm text-left 2xl:text-right">
+            OUSD total supply ÷ Rebasing OUSD supply
```
[the remark this answers is not in the document]
Comment (HrikB) [reply components/proof-of-yield/YieldBoostMultiplier.tsx:44]: Yup. Also for a component on page not yet implemented (See https://www.ousd.com/proof-of-yield/1675886432000)

File: hooks/index.ts (hunk 5 above)
Comment (HrikB) [reply hooks/index.ts:7]: I didn't know about this... but def worth exploring when we're resolving tech debt. I see many of these custom hooks could be replaced with what the library provides.

File: sections/proof-of-yield/DailyYield.tsx (hunk 6 above)
Comment (HrikB) [reply sections/proof-of-yield/DailyYield.tsx:110]: The point of the wrappers is mostly to be able to not have to duplicate the styling. Initially, I used the `td` and `th` elements directly and just had `[element]ClassName` strings at the top of the file but we decided it was better to just make it into a component.

File: sections/proof-of-yield/DayOtherSources.tsx [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  sectionOverrideCss?: string;
+}
+
+const DayOtherSources = ({ sectionOverrideCss }: DayOtherSourcesProps) => {
+  return (
+    <Section className={twMerge("", sectionOverrideCss)}>
+      <Table className="mt-8">
+        <thead>
+          <tr>
+            <TableHead align="left" className="pl-4 md:pl-8 w-auto">
```
[the remark this answers is not in the document]
Comment (HrikB) [reply sections/proof-of-yield/DayOtherSources.tsx:35]: Kind of same response to "As Im looking at this usage, I dont think we should really even create a wrapper for td or th, might as well just add the elements directly"

The `table`s throughout PoY had similar styling requirements but the elements within them necessarily didn't. I initially had tried to remove it from the higher level components but the elements within each table were too different.

File: hooks/index.ts (hunk 5 above)
Comment (HrikB) [reply hooks/index.ts:7]: I didn't know about this... but def worth exploring when we're resolving tech debt. I see many of these custom hooks could be replaced with what the library provides.

File: components/proof-of-yield/TableHead.tsx (hunk 4 above)
Comment (HrikB) [reply components/proof-of-yield/TableHead.tsx:20]: Made this change but `text-`doesn't always work if what I'm aligning is a component. So used `float-` in that case.

**Review by smitch88 — APPROVED** [review]

**Pull request merged**